_dev/src/ts/routing/ScriptHandler.ts

```diff
@@ -60,44 +65,56 @@ export default class ScriptHandler {
 
   /**
    * @public
-   * @param {string} scriptID - The ID of the route to load his associated script.
+   * @param {string} scriptID - The ID of the script to load.
    * @returns void
-   * @description Loads and mounts the page script associated with the specified route name.
+   * @description Loads and mounts the script associated with the specified script name.
    */
-  loadScript(scriptID: string) {
-    const classScript = this.#routesMatching[scriptID];
-    if (classScript) {
-      try {
-        this.#currentScript = new classScript();
-        this.#currentScript.mount();
-      } catch (error) {
-        console.error(`Failed to load script with ID ${scriptID}:`, error);
+  public loadScript(scriptID: string) {
+    const scriptType = this.#getScriptTypeByScriptID(scriptID);
+
+    if (!scriptType) {
+      return;
+    }
+
+    const classScript = this.#scriptsMatching[scriptType][scriptID];
+
+    try {
+      if (this.#currentScripts[scriptType] !== undefined) {
+        this.unloadScript(scriptType);
       }
-    } else {
-      console.debug(`No matching class found for ID: ${scriptID}`);
+      this.#currentScripts[scriptType] = new classScript();
+      this.#currentScripts[scriptType].mount();
+    } catch (error) {
+      console.error(`Failed to load script with ID ${scriptID}:`, error);
     }
   }
 
   /**
    * @public
-   * @param {string} newRoute - The name of the route to load his associated script.
+   * @param {ScriptType} scriptType - The type of the script to unload his associated script.
    * @returns void
-   * @description Updates the currently loaded route script by destroying the current
-   *              page instance and loading a new one based on the provided route name.
+   * @description Unloads the currently loaded script from his type.
+   *  Should be called before updating the DOM.
    */
-  public updateRouteScript(newRoute: string) {
-    this.#currentScript?.beforeDestroy();
-    this.loadScript(newRoute);
+  public unloadScript(scriptType: ScriptType): void {
```

Comment:
```suggestion
  public unloadScriptType(scriptType: ScriptType): void {
```

To avoid the confusion with `loadScript` which receives a script name.